modules/test/playwright/tests/portal-workflow-kaleo-designer-web/timerNotification.spec.ts

```diff
@@ -0,0 +1,115 @@
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+import {Page, expect, mergeTests} from '@playwright/test';
+
+import {apiHelpersTest} from '../../fixtures/apiHelpersTest';
+import {applicationsMenuPageTest} from '../../fixtures/applicationsMenuPageTest';
+import {loginTest} from '../../fixtures/loginTest';
+import {workflowPagesTest} from '../../fixtures/workflowPagesTest';
+import {ActionNotificationPage} from '../../pages/portal-workflow-kaleo-designer-web/ActionNotificationPage';
+import {getRandomInt} from '../../utils/util';
+
+export const test = mergeTests(
+	apiHelpersTest,
+	applicationsMenuPageTest,
+	loginTest,
+	workflowPagesTest
+);
+
+const timerNotifications = [
+	{
+		notificationDescription: 'notificationDescription0' + getRandomInt(),
+		notificationName: 'notificationName0' + getRandomInt(),
+		notificationTypeEmail: true,
+		notificationTypeUser: true,
+		recipientType: 'role',
+		recipientTypeData: {
+			roleName: 'Account Manager',
+		},
+		template: 'template0' + getRandomInt(),
+		templateLanguage: 'freemarker',
+	},
+	{
+		notificationDescription: 'notificationDescription1' + getRandomInt(),
+		notificationName: 'notificationName1' + getRandomInt(),
+		notificationTypeEmail: true,
+		notificationTypeUser: true,
+		recipientType: 'scriptedRecipient',
+		recipientTypeData: {
+			script: 'script' + getRandomInt(),
+			scriptLanguage: 'java',
+		},
+		template: 'template1' + getRandomInt(),
+		templateLanguage: 'text',
+	},
+];
+test('LPD-16281 can create timer notifications', async ({
+	page,
+	workflowDefinitionPage,
+}) => {
+	await workflowDefinitionPage.goto();
+	await workflowDefinitionPage.clickSingleAproverWorkflowDefinition();
+	await workflowDefinitionPage.clickReviewNodeLink();
+	await workflowDefinitionPage.createTimerNotification(timerNotifications);
+	await workflowDefinitionPage.updateWorkflowDefinition();
+	await workflowDefinitionPage.goBack();
+	await workflowDefinitionPage.clickSingleAproverWorkflowDefinition();
+	await workflowDefinitionPage.clickReviewNodeLink();
+
+	const timerOption = workflowDefinitionPage.page.getByRole('link', {
+		name: 'Duration: 3 week',
+	});
+
+	await expect(timerOption).toBeVisible();
+
+	await timerOption.click();
+
+	await assertActionNotification(0, page, timerNotifications[0]);
+	await assertActionNotification(1, page, timerNotifications[1]);
+
+	// clean up
+
+	await workflowDefinitionPage.deleteAllTimers();
+	await workflowDefinitionPage.updateWorkflowDefinition();
+});
+
+async function assertActionNotification(
+	index: number,
+	page: Page,
+	timerNotification: DataObject
+) {
+	const {
+		inputNotificationDescription,
+		inputNotificationName,
+		inputNotificationTemplate,
+		inputNotificationTemplateLanguage,
+		inputRecipientType,
+		inputRoleName,
+		inputScript,
+		inputScriptLanguage,
+	} = new ActionNotificationPage(page, index);
+
+	await expect(inputNotificationDescription).toBeVisible();
```

Comment:
Assertions should be against the value. Check the value, not only if the element is visible. Something like this:

```
await expect(inputNotificationDescription).toHaveValue(
		timerNotifications[index].notificationDescription
	);

	await expect(inputNotificationName).toHaveValue(
		timerNotifications[index].notificationName
	);

	await expect(inputNotificationTemplate).toHaveValue(
		timerNotifications[index].template + "\n"
	);

	await expect(inputNotificationTemplateLanguage).toHaveValue(
		timerNotifications[index].templateLanguage
	);

	await expect(page.getByText(/Email/).nth(index)).toBeVisible();

	await expect(page.getByText(/User Notification/).nth(index)).toBeVisible();

	await expect(inputRecipientType).toHaveValue(
		timerNotifications[index].recipientType
	);

	if (timerNotification.recipientType === 'role') {
		await expect(inputRoleName).toHaveValue(
			timerNotifications[index].recipientTypeData.roleName
		);
	} else if (timerNotification.recipientType === 'scriptedRecipient') {
		await expect(inputScript).toHaveValue(
			timerNotifications[index].recipientTypeData.script + "\n"
		);
		await expect(inputScriptLanguage).toHaveValue(
			timerNotifications[index].recipientTypeData.scriptLanguage
		);
	}

```

Reply:
But there is a problem with the above: Playwright is failing to properly selecting "java" `scriptLanguage` in Scripted Recipient recipient type, in `ActionNotificationPage.fillActionNotificationFields`. Because of this, `toHaveValue` is failing here.